### PR TITLE
[Doc] Add Data Structures and Elements

### PR DESCRIPTION
**Pull request by MengzhangLI**

### CODE REVIEW

File: docs/en/advanced_guides/structures.md [hunk 1]
```diff
@@ -1 +1,162 @@
...
+for img, img_metas, rescale in dataloader:
+  seg_logit = whole_inference(img, img_meta, rescale)
+
+# Inference on sliding window
+for img, img_metas, rescale in dataloader:
+  seg_logit = slide_inference(img, img_meta, rescale)
+```
+
+From the above code examples, we can see that without encapsulation, the different data required by different tasks and algorithms lead to inconsistent interfaces between their modules, which seriously affects the extensibility and reusability of the library.
+Therefore, in order to solve the above problem, we use [DataElement tutorial](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/advanced_tutorials/data_element.md) from MMEngine to encapsulate the data required for each task into `data_sample`.
```
Comment (MeowZheng) [inline docs/en/advanced_guides/structures.md:20]: please change to English documentation

File: docs/en/advanced_guides/structures.md [hunk 2]
```diff
@@ -1 +1,162 @@
-# Structures
+# Data Structures and Elements
```
Comment (MeowZheng) [inline docs/en/advanced_guides/structures.md:1]: Why is there 'Elements' in title?

File: docs/en/advanced_guides/structures.md [hunk 3]
```diff
@@ -1 +1,162 @@
...
+
+During the training/testing process of a model, there is often a large amount of data to be passed between modules, and the data required by different tasks or algorithms is usually different. For example, in MMSegmentation, in training it needs image and its meta information but in inference it also needs rescaling information of input images. This makes the interfaces of different tasks or models may be inconsistent:
```
Comment (MeowZheng) [inline docs/en/advanced_guides/structures.md:3]: in training -> during training

what is rescaling information?

File: docs/en/advanced_guides/structures.md [hunk 4]
```diff
@@ -1 +1,105 @@
 # Structures
+
+To unify input and output interfaces  between different models and modules, OpenMMLab 2.0 MMEngine defines an abstract data structure,
+it has implemented basic functions of `Create`, `Read`, `Update`, `Delete`, supported data transferring among different types of devices
+and tensor-like or dictionary-like operations such as `.cpu()`, `.cuda()`, `.get()` and `.detach()`.
+More details can be found [here](https://github.com/open-mmlab/mmengine/blob/main/docs/en/advanced_tutorials/data_element.md).
+
+Similarly, MMSegmentation also follows this data structure requirement. SegDataSample is used to encapsulate the data needed
```
Comment (MeowZheng) [inline docs/en/advanced_guides/structures.md:8]:
```suggestion
MMSegmentation also follows this interface protocol and defines `SegDataSample` which is used for encapsulate the data of semantic segmentation task
```

File: docs/en/advanced_guides/structures.md [hunk 5]
```diff
@@ -1 +1,104 @@
...
+
+## Semantic Segmentation Data SegDataSample
+
+[SegDataSample](mmseg.structures.SegDataSample) includes three main fields `gt_sem_seg`, `pred_sem_seg` and `seg_logits`, which are used to store the annotation information and prediction results respectively.
+
+| Field          | Type                      | Description                     |
+| -------------- | ------------------------- | ------------------------------- |
+| gt_sem_seg     | [`PixelData`](#pixeldata) | Annotation information.         |
+| pred_instances | [`PixelData`](#pixeldata) | The predicted result.           |
+| seg_logits     | [`PixelData`](#pixeldata) | The logits of predicted result. |
```
Comment (MeowZheng) [inline docs/en/advanced_guides/structures.md:18]:
```suggestion
| seg_logits     | [`PixelData`](#pixeldata) | The model output before normalized |
```

File: docs/en/advanced_guides/structures.md [hunk 6]
```diff
@@ -1 +1,104 @@
...
+
+## Semantic Segmentation Data SegDataSample
+
+[SegDataSample](mmseg.structures.SegDataSample) includes three main fields `gt_sem_seg`, `pred_sem_seg` and `seg_logits`, which are used to store the annotation information and prediction results respectively.
+
+| Field          | Type                      | Description                     |
+| -------------- | ------------------------- | ------------------------------- |
+| gt_sem_seg     | [`PixelData`](#pixeldata) | Annotation information.         |
+| pred_instances | [`PixelData`](#pixeldata) | The predicted result.           |
+| seg_logits     | [`PixelData`](#pixeldata) | The raw (non-normalized) predicted result. |
```
Comment (MeowZheng) [inline docs/en/advanced_guides/structures.md:18]:
```suggestion
| Field          | Type                      | Description                                |
| -------------- | ------------------------- | ------------------------------------------ |
| gt_sem_seg     | [`PixelData`](#pixeldata) | Annotation information.                    |
| pred_instances | [`PixelData`](#pixeldata) | The predicted result.                      |
| seg_logits     | [`PixelData`](#pixeldata) | The raw (non-normalized) predicted result. |
```
